src/app/tests/suites/TestGroupsCluster.yaml

```diff
@@ -0,0 +1,304 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Groups Cluster Tests
+
+config:
+    cluster: "Groups"
+    endpoint: 1
+
+tests:
+    - label: "View Group 0 (invalid)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0
+      response:
+          values:
+              - name: "groupId"
+                value: 0
+              - name: "status"
+                value: 0x87
+
+    - label: "View Group 1 (not found)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x0001
+      response:
+          values:
+              - name: "groupId"
+                value: 0x0001
+              - name: "status"
+                value: 0x8B
+
+    - label: "Add Group 1 (new)"
+      command: "AddGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x0001
+              - name: "groupName"
+                value: "Group #1"
+      response:
+          values:
+              - name: "groupId"
+                value: 0x0001
+              - name: "status"
+                value: 0
+
+    - label: "View Group 1 (new)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x0001
+      response:
+          values:
+              - name: "groupId"
+                value: 0x0001
+              - name: "status"
+                value: 0
```

Comment:
The names where removed from the tests due to the missing zero string terminator. Since that problem is now fixed, I put back the names in the tests.